### PR TITLE
[#5088] Add annotation which marks packages/interfaces/classes as uns…

### PR DESCRIPTION
**Pull request by normanmaurer**
…table

Motivation:

Some codecs should be considered unstable as these are relative new. For this purpose we should introduce an annotation which these codecs should us to be marked as unstable in terms of API.

Modifications:

- Add UnstableApi annotation and use it on codecs that are not stable
- Move http2.hpack to http2.internal.hpack as it is internal.

Result:

Better document unstable APIs.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@buchgr @trustin PTAL

File: common/src/main/java/io/netty/util/internal/UnstableApi.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import java.lang.annotation.Target;
+
+/**
+ * Indicates a public API that can change at any time (even in minor/bugfix releases).
+ *
+ * Usage guidelines:
+ *
+ * <ol>
+ *     <li>Is not needed for things located in *.internal.* packages</li>
+ *     <li>Only public accessable classes/interfaces must be annotated</li>
```
Comment (buchgr) [inline common/src/main/java/io/netty/util/internal/UnstableApi.java]: s/accessable/acessible/

**Comment by buchgr** [comment]
Overall LGTM. I didn't check if any class is missing or all the right classes have been marked unstable, because quite frankly I don't know :-).

**Comment by normanmaurer** [comment]
@trustin OK to merge ?

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 as 9229ed98e2eb3159abe3195459e0ffc78200d358

**Pull request closed without merging**